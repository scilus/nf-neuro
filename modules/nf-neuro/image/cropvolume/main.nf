
process IMAGE_CROPVOLUME {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(image), path(bounding_box)

    output:
    tuple val(meta), path("*_cropped.nii.gz"), emit: image
    tuple val(meta), path("*.pkl")           , emit: bounding_box, optional: true
    path "versions.yml"                      , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    def input_bbox = bounding_box ? "--input_bbox $bounding_box" : ""
    def suffix = task.ext.first_suffix ? "${task.ext.first_suffix}_cropped" : "cropped"
    def output_bbox = task.ext.output_bbox ? "--output_bbox ${prefix}_${suffix}_bbox.pkl" : ""

    """
    scil_volume_crop.py $image ${prefix}_${suffix}.nii.gz $input_bbox $output_bbox

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.first_suffix ? "${task.ext.first_suffix}_cropped" : "cropped"

    """
    scil_volume_crop.py -h

    touch ${prefix}_${suffix}.nii.gz

    if $task.ext.output_bbox;
    then
        touch ${prefix}_${suffix}_bbox.pkl
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
