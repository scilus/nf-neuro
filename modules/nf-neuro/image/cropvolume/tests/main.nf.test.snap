{
    "image - cropvolume": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,c9f9039fc9103e716e831016a61807f9"
                ],
                "bounding_box": [
                    
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c9f9039fc9103e716e831016a61807f9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-12T18:59:41.317523464"
    },
    "image - cropvolume - outputbbox": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,d6a03b61669b07683e8c4b21a613f16c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c9f9039fc9103e716e831016a61807f9"
                ],
                "bounding_box": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped_bbox.pkl:md5,d6a03b61669b07683e8c4b21a613f16c"
                    ]
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_cropped.nii.gz:md5,93cc10f69b0829409c69bf743e9c95ef"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c9f9039fc9103e716e831016a61807f9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-12T18:59:50.080977383"
    }
}